### DISCUSSION TITLE
FAQ about codebase an contributing

### DISCUSSION THREAD

**Discussion by mh-alahdadian** [react-hook-form/react-hook-form#2074] [General]
I think that there should be such topic to asks around code base and help contributors to make PR without problem

**Comment by mh-alahdadian** [comment]
about options of `useForm` shouldn't there some effects around its field to update states with new options?
do we consider options param to be a simple object and never changes it's properties in different renders?

**Reply by bluebill1049** [reply]
can you explain on this a bit more? You mean an object to merge with final submit data?

**Reply by mh-alahdadian** [reply]
I'm try to say that defaultValues should not be changed in different renders and we should tell user to not change options
or if we want to let users change default values (or other part of options object) maybe we need to edit values of form after changing of defaultValues (not touched fields only)

**Comment by mh-alahdadian** [comment]
should we write a class in some cases where it's make sense ?
like write a new Control class instead of createControl function

**Reply by bluebill1049** [reply]
as long as we can reduce the code weight, I am all for it 👀

**Comment by mh-alahdadian** [comment]
how do watch works with uncontrolled inputs and specially with custom uncontrolled inputs which has only name and value fields and no onchange?
